sql/python/sqlflow_submitter/maxcompute.py

```diff
@@ -19,19 +20,67 @@ class MaxCompute:
     @staticmethod
     def connect(database, user, password, host):
         return ODPS(user, password, project=database, endpoint=host)
-
+    
     @staticmethod
-    def execute(conn, statement):
-        compress = tunnel.CompressOption.CompressAlgorithm.ODPS_ZLIB
-        inst = conn.execute_sql(statement)
-        if not inst.is_successful():
-            return None, None
-        r = inst.open_reader(tunnel=True, compress_option=compress)
-        field_names = [col.name for col in r._schema.columns]
-        rows = [[v[1] for v in rec] for rec in r[0: r.count]]
-        return field_names, list(map(list, zip(*rows))) if r.count > 0 else None
-
+    def db_generator(conn, statement, feature_column_names,
+            label_column_name, column_name_to_type, fetch_size):
+        def reader():
+            compress = tunnel.CompressOption.CompressAlgorithm.ODPS_ZLIB
+            inst = conn.execute_sql(statement)
+            if not inst.is_successful():
+                return None
+    
+            r = inst.open_reader(tunnel=True, compress_option=compress)
+            field_names = None if r._schema.columns is None \
+                    else [col.name for col in r._schema.columns]
+            label_idx = field_names.index(label_column_name)
+    
+            i = 0
+            while i < r.count:
+                expected = r.count-i if r.count-i < fetch_size else fetch_size
+                for row in [[v[1] for v in rec] for rec in r[i: i+expected]]:
```

Comment:
Just curious about is indexing reader like `r[i: i+expected]` read only a part from the table?

Reply:
The iterator in the loop calls [function: read](https://github.com/aliyun/aliyun-odps-python-sdk/blob/master/odps/models/instance.py#L808) which returns a generator here to read records on demand.